cpp/src/arrow/json/chunked_builder.cc

```diff
@@ -224,16 +224,16 @@ class ChunkedListArrayBuilder : public ChunkedArrayBuilder {
   Status Finish(std::shared_ptr<ChunkedArray>* out) override {
     RETURN_NOT_OK(task_group_->Finish());
 
-    std::shared_ptr<ChunkedArray> child_array;
-    RETURN_NOT_OK(value_builder_->Finish(&child_array));
+    std::shared_ptr<ChunkedArray> value_array;
+    RETURN_NOT_OK(value_builder_->Finish(&value_array));
 
-    auto type = list(value_field_->WithType(child_array->type()));
+    auto type = list(value_field_->WithType(value_array->type())->WithMetadata(nullptr));
```

Comment:
So it's essentially `list(field(value_field_->name(), value_array->type()))`?

Reply:
@pitrou Yes, I'd prefer to leave this for follow up since CI is already green